File: web/src/app/models/tarefa.ts
export type TarefaStatus = 'Pendente' | 'Concluída';

export interface Tarefa {
  id: number;
  titulo: string;
  descricao: string | null;
  status: TarefaStatus;
  dataCriacao: string;
}

export interface TarefaUpsert {
  titulo: string;
  descricao: string | null;
  status: TarefaStatus;
}

File: api/Models/Tarefa.cs
using System.ComponentModel.DataAnnotations;

namespace ImpactaTech.Api.Models;

public class Tarefa
{
    public int Id { get; set; }

    [Required]
    public string Titulo { get; set; } = string.Empty;

    public string? Descricao { get; set; }

    [Required]
    public string Status { get; set; } = "Pendente";

    public DateTime DataCriacao { get; set; }
}
